### PR TITLE
import-idb: Add tests for IDA files

### PR DESCRIPTION
**Pull request by dwarfyresearch**

### CODE REVIEW

File: share/revng/test/configuration/revng-qa/for-import-idb.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+tags:
+- name: ida
+  variables:
+    IFLAGS:
+      - -A
+      - -B
```
Comment (aleclearmind) [inline share/revng/test/configuration/revng-qa/for-import-idb.yml]: I'd inline those unless you foresee reasons to configure these.

File: share/revng/test/configuration/revng-qa/for-import-idb.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+tags:
+- name: ida
```
Comment (aleclearmind) [inline share/revng/test/configuration/revng-qa/for-import-idb.yml]: I think we've been using `for-ida` as a pattern for tags.

File: share/revng/test/tests/model/import/idb/enum-type.c [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+
+enum A {
+   x,
+   y
+};
+
+void fn(enum A x)
+{
```
Comment (aleclearmind) [inline share/revng/test/tests/model/import/idb/enum-type.c]: Can you run `revng-check-conventions`? It should work.

File: share/revng/test/configuration/revng-qa/for-import-idb.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  - tests/model/import/idb/struct.c
+  - tests/model/import/idb/typedef-char-to-uint32.c
+commands:
+- type: revng-qa.idb
+  from:
+  - type: revng-qa.compiled
+    filter: ida
+  suffix: .i64
+  command: |-
+    unshare -rn ida64 -o$OUTPUT $IFLAGS $INPUT
```
Comment (aleclearmind) [inline share/revng/test/configuration/revng-qa/for-import-idb.yml]: So this works assuming `ida64` is in `PATH`, correct?

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: share/revng/test/configuration/revng-qa/for-import-idb.yml (hunk 4 above)
Comment (dwarfyresearch) [reply share/revng/test/configuration/revng-qa/for-import-idb.yml]: Yes... I am not sure how are we going to ensure that.

File: share/revng/test/tests/model/import/idb/enum-type.c (hunk 3 above)
Comment (aleclearmind) [reply share/revng/test/tests/model/import/idb/enum-type.c]: Ping

File: share/revng/test/configuration/revng-qa/for-import-idb.yml (hunk 4 above)
Comment (aleclearmind) [reply share/revng/test/configuration/revng-qa/for-import-idb.yml]: I'll take care of this.

File: share/revng/test/tests/model/import/idb/enum-type.c (hunk 3 above)
Comment (dwarfyresearch) [reply share/revng/test/tests/model/import/idb/enum-type.c]: It was addressed back-then with the commit bellow, I have just forgotten to mark it as `resolved`. Sorry.

File: share/revng/test/configuration/revng-qa/for-import-idb.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  - tests/model/import/idb/struct.c
+  - tests/model/import/idb/typedef-char-to-uint32.c
+commands:
+- type: revng-qa.idb
+  from:
+  - type: revng-qa.compiled
+    filter: for-ida
+  suffix: .i64
+  command: |-
+    unshare -rn ida64 -o$OUTPUT -A -B $INPUT
```
Comment (aleclearmind) [inline share/revng/test/configuration/revng-qa/for-import-idb.yml]:
```suggestion
    idat64 -o$OUTPUT -A -B $INPUT
    rm "$$(echo $OUTPUT | sed 's/\.i64/.asm/')"
```

**Review by aleclearmind — CHANGES_REQUESTED** [review]

**Pull request closed without merging**